### ISSUE TITLE
[2021.09] PHP Notices

### ISSUE THREAD

**Issue by tobiasd** [friendica/friendica#10474]
Please report here any PHP Notice message obtained running the 2021.01 stable or 2021.06-dev versions.

Closes #10169

**Comment by ghost** [comment]
Latest develop

```
PHP Notice:  Undefined index: uri in /src/Model/APContact.php on line 355
PHP Notice:  Undefined index: uri in /src/Model/Contact.php on line 2160
```
Addressed in #10478

**Comment by ghost** [comment]
Latest dev

`PHP Notice:  Undefined index: nick in /src/Model/Contact.php on line 2176`

Fixed by https://github.com/friendica/friendica/pull/10489

**Comment by ghost** [comment]
latest develop
```

PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 322
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 328 
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 331
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 332
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 333
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 334
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 339
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 340
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 342
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 351
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 352
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 353
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 354
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 355
PHP Notice:  Trying to access array offset on value of type bool in a/src/Protocol/ActivityPub/Transmitter.php on line 356
PHP Notice:  Trying to access array offset on value of type bool in /src/Protocol/ActivityPub/Transmitter.php on line 358
```
Addressed by #10525

**Comment by AlfredSK** [comment]
2021.09-dev

`Trying to access array offset on value of type null in .../src/Object/Thread.php on line 86`

Addressed by https://github.com/friendica/friendica/pull/10543

**Comment by 4nanook** [comment]
Getting a lot of these on 2021.06-dev:

[Sun Jun 13 15:39:21.458563 2021] [:error] [pid 417251:tid 139753236182784] [client 193.202.84.190:33571] PHP Notice:  Undefined index: post-type in /misc/www/friendica/src/Model/Item.php on line 2805

<strike>Addressed by #10704</strike>

**Comment by AlfredSK** [comment]
2021.09-dev

```
Undefined index: theme in .../src/Model/Profile.php on line 227
Undefined index: picdate in .../src/Model/Profile.php on line 289
```
Addressed in #10628

**Comment by ghost** [comment]
`PHP Notice:  Undefined variable: result in /addon/twitter/twitter.php on line 527`

Addressed in  https://github.com/friendica/friendica-addons/pull/1169

**Comment by nupplaphil** [comment]
```console
Undefined index: hide-friends in /var/www/html/src/Model/Profile.php on line 402
Undefined index: hide-friends in /var/www/html/src/Module/Profile/Contacts.php on line 61
```
latest dev

Addressed in #10628

**Comment by tobiasd** [comment]
This was reported in the XMPP support channel. It occurs right after the switch from 2021.07 stable to 2021.09-rc

```
PHP Notice:  Trying to access array offset on value of type bool in /src/Database/Database.php on line 1465
```

Addressed by #10704

**Comment by 4nanook** [comment]
Friendica just spit out three PHP notices:
```
PHP Notice:  Undefined index: uri-id in /misc/www/friendica/src/Worker/Notifier.php on line 538                                                                                               
PHP Notice:  Undefined index: uri-id in /misc/www/friendica/src/Worker/Notifier.php on line 540                                                                                               
PHP Notice:  Undefined index: uid in /misc/www/friendica/src/Worker/Notifier.php on line 540                                                                                                  
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Friendica\Model\Item::storeForUserByUriId() must be of the type int, null given, called in                                         
/misc/www/friendica/src/Worker/Notifier.php on line 540 and defined in /misc/www/friendica/src/Model/Item.php:1365                                                                            
Stack trace:                                                                                                                                                                                  
#0 /misc/www/friendica/src/Worker/Notifier.php(540): Friendica\Model\Item::storeForUserByUriId()                                                                                              
#1 /misc/www/friendica/src/Worker/Notifier.php(488): Friendica\Worker\Notifier::delivery()                                                                                                    
#2 [internal function]: Friendica\Worker\Notifier::execute()                                                                                                                                  
#3 /misc/www/friendica/src/Core/Worker.php(458): call_user_func_array()                                                                                                                       
#4 /misc/www/friendica/src/Core/Worker.php(348): Friendica\Core\Worker::execFunction()                                                                                                        
#5 /misc/www/friendica/src/Core/Worker.php(111): Friendica\Core\Worker::execute()                                                                                                             
#6 /misc/www/friendica/bin/worker.php(83): Friendica\Core\Worker::processQueue()                                                                                                              
#7 {main}                                                                                                                                                                                     
  thrown in /misc/www/friendica/src/Model/Item.php on line 1365          
```
Version
    Friendica 'Siberian Iris' 2021.09-dev - 1424 
Actually it keeps throwing these now, I haven't changed anything.  

 PHP 7.4.23 (cli) (built: Aug 26 2021 15:51:37) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.23, Copyright (c), by Zend Technologies

mysql  Ver 15.1 Distrib 10.5.12-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

There were a bunch of updates to PHP yesterday.

**Comment by 4nanook** [comment]
I applied any outstanding updates today and strangely the database number DECREASED, so I'm a bit confused:

This is Friendica, version 2021.09-dev that is running at the web location https://friendica.eskimo.com. The database version is 1434/1434, the post update version is 1427/1423.

The error went away.

**Comment by AlfredSK** [comment]
The database version seems to be fine. The postupdate version is weird. Can you please watch if the second number is changing after some time?

> the post update version is 1427/**1423**

The thing is it should be 1423/1427 and it should go up to 1427/1427 when all post updates were applied.

I'M asking because of this:  #10639

**Comment by 4nanook** [comment]
It isn't changing, is there something I can do manually to force the 
issue?

-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
  Eskimo North Linux Friendly Internet Access, Shell Accounts, and Hosting.
    Knowledgeable human assistance, not telephone trees or script readers.
  See our web site: http://www.eskimo.com/ (206) 812-0051 or (800) 246-6874.

**Comment by AlfredSK** [comment]
Yes, there is a command to run it manually. The thing is it should be applied automatically. If it doesn't work we have a bug and should find the cause of it.

**Comment by 4nanook** [comment]
Then you have a bug, my database is very fast, it is on flash memory,
and has a lot of RAM dedicated as well so it can buffer most things.  These
updates normally only take a few seconds here.

      What is the command I need to run?

-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
  Eskimo North Linux Friendly Internet Access, Shell Accounts, and Hosting.
    Knowledgeable human assistance, not telephone trees or script readers.
  See our web site: http://www.eskimo.com/ (206) 812-0051 or (800) 246-6874.

**Comment by AlfredSK** [comment]
You can have a look at the linked issue. I posted it there.

**Comment by ghost** [comment]
latest rc

`PHP Notice:  Undefined variable: result in /addon/twitter/twitter.php on line 527`

Addressed by https://github.com/friendica/friendica-addons/pull/1169

**Comment by AlfredSK** [comment]
```
Undefined index: rel in .../src/Model/Contact.php on line 2762
Undefined index: rel in .../src/Model/Contact.php on line 2762
Undefined index: id in .../src/Model/Contact.php on line 2765
```

And...

`Trying to access array offset on value of type bool in .../src/Module/Contact.php on line 296`

2021.09-rc PHP 7.4

Addressed by #10704

**Comment by AlfredSK** [comment]
2021.09-rc PHP 7.4

```
Trying to access array offset on value of type bool in .../src/Model/Mail.php on line 80
Trying to access array offset on value of type bool in .../src/Model/Mail.php on line 81
```
Addressed in #10764

**Comment by AlfredSK** [comment]
2021.09-rc PHP 7.4

```
Trying to access array offset on value of type bool in .../src/Protocol/OStatus.php on line 1942
Trying to access array offset on value of type bool in .../src/Protocol/OStatus.php on line 1943
```
Addressed in #10764
